### ISSUE TITLE
Form: Focus not moved to error field after activate save button &  Screen reader does not announce error message after click on save button.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2217]
[A11Y-4091](https://jira.cengage.com/browse/A11Y-4091) ✅
[A11Y-4092](https://jira.cengage.com/browse/A11Y-4092)

**Issue 1:**
**Focus not moved to error field after activate save button**

Platform:
Windows 11 / Chrome /Keyboard

Steps to reproduce:
1. Open the URL "https://faculty-stage.cengage.com/"
2. Click on "sign in" button
3. On sign in page enter instructor email and password and do sign in
4. Select "My course" from main menu
5. Select course key: MTPQGMDZD55G
6. Navigate near "Bulk Edit" and activate it.
7.Select the content and move near "Edit X selected" button
8.Expand the button, and select "Apply Setting Template" and expand it.
9.Navigate near save button and click it.
Observe that error message displayed, but focus does not move to error field.

Actual Result:
After click save button, error appear for "activity setting template" field, but focus remains on save button instead of shift to "Activity setting template" field.

Note: Same issue is repro for "edit dates" dialog.

Expected Result:
After activate save button focus should move to error field.

WCAG Guideline:
2.4.3 Focus order

User Impact:
Keyboard-only users may not realize an error has occurred or where it is located, especially on long or complex forms.

Recommendation:
On validation failure, programmatically move focus to the first invalid field and Visually highlight the error and ensure focus indicators are clear.

Screencast:
https://app.screencast.com/Ed44tE5EvQ20t

**Issue 2:**
**Screen reader does not announce error message after click on save button.**

Platform:
Windows 11 / Edge 136 / NVDA 2025
Chrome / JAWS 2025
Mac15/ Safari/ Voiceover

Steps to reproduce:
1. Open the URL "https://faculty-stage.cengage.com/"
2. Click on "sign in" button
3. On sign in page enter instructor email and password and do sign in
4. Select "My course" from main menu
5. Select course key: MTPQGMDZD55G
6. Navigate near "Bulk Edit" and activate it.
7.Select the content and move near "Edit X selected" button
8.Expand the button, and select "Apply Setting Template" and expand it.
9.Navigate near save button and click it.
Observe that error message displayed, and screen reader remains silent.

Actual Result:
After click save button, error appear for "activity setting template" field and screen reader remains silent instead of announce the error message "Select settings template to apply".

Note: Same issue is repro for "edit dates" dialog.

Expected Result:
Screen reader should announce the error message such as "unable to procees please select settings template to apply"

WCAG Guideline:
4.1.3 – Status Messages

User Impact:
Screen reader users may not realize a validation error has occurred.
This may lead to: Form submission failure,
Missed required fields and Frustration or abandonment of the task
This creates a barrier for users who are blind or visually impaired, and fails accessibility compliance.

Recommendation:
Use ARIA live regions to announce error messages immediately.

Screencast:
https://app.screencast.com/nUvttm1aZ8eZl

**Comment by silvalaura** [comment]
No Magma changes required. @olehnoskov-cengage to inform MT team how to address on their end.

**Comment by olehnoskov-cengage** [comment]
Added ready to use fix in the comments -> https://codesandbox.io/p/devbox/react-magma-example-forked-gvt7f3?file=%2Fexample.tsx&workspaceId=ws_8SENM8VWD7GtdwUe2iH82S
